lib/mongoid/criteria.rb

```diff
@@ -55,6 +57,40 @@ def ==(other)
       entries == other
     end
 
+    # Call either Findable#find or Enumerable#find based on the given arguments
+    #
+    # If the arguments match Enumerable#find signature, that will be invoked.
+    # Otherwise Findable#find will be invoked
+    #
+    # The signature of Enumerable#find:
+    #
+    #   find(ifnone = nil) { |obj| block } → obj or nil
+    #   find(ifnone = nil) → an_enumerator
+    #
+    # @example Find using a block, invokes Enumerable#find
+    #   criteria.find { |item| item.name == "Depeche Mode" }
+    #
+    # @example Find given a block and a Proc, invokes Enumerable#find
+    #   criteria.find(-> { "Default Band" }) { |item| item.name == "Milwaukee Mode" }
+    #
+    # @example Find given just a Proc, invokes Enumerable#find, returns an Enumerator
+    #   enumerator = criteria.find(-> { "Default Band" })
+    #
+    # @example Find given anything else, invokes Findable#find
+    #   critera.find("1234")
+    #
+    # @return Same return value as either Findable#find or Enumerable#find, depending
+    #   on which one was invoked.
+    #
+    # @since 7.1.1
```

Comment:
We no longer use since tags, and this one isn't accurate since the change would go into the next major release.

Reply:
removed at 397f4c2b9